File: test/cs/case30_slice_basic.cs
using System.Collections.Generic;
using System.IO;
using System;

public static class Program
{
    public static void main()
    {
        List<long> nums = new List<long> { 10, 20, 30, 40, 50 };
        string text = "abcdef";
        List<long> mid_nums = Pytra.CsModule.py_runtime.py_slice(nums, (long?)(1), (long?)(4));
        string mid_text = Pytra.CsModule.py_runtime.py_slice(text, (long?)(2), (long?)(5));
        Pytra.CsModule.py_runtime.print(mid_nums[0]);
        Pytra.CsModule.py_runtime.print(mid_nums[2]);
        Pytra.CsModule.py_runtime.print(mid_text);
    }

    public static void Main(string[] args)
    {
        main();
    }
}

File: test/rs/case30_slice_basic.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

fn main() -> () {
    let mut nums: Vec<i64> = vec![10, 20, 30, 40, 50];
    let mut text: String = "abcdef".to_string();
    let mut mid_nums: Vec<i64> = py_slice(&(nums), Some(1), Some(4));
    let mut mid_text: String = py_slice(&(text), Some(2), Some(5));
    py_print((mid_nums)[0 as usize]);
    py_print((mid_nums)[2 as usize]);
    py_print(mid_text);
}
